doc/lightningd-config.5.md

```diff
@@ -103,9 +103,28 @@ Sets the working directory. All files (except *--conf* and
  **pid-file**=*PATH*
 Specify pid file to write to.
 
- **log-level**=*LEVEL*
+ **log-level**=*LEVEL*\[:*SUBSYSTEM*\]
 What log level to print out: options are io, debug, info, unusual,
-broken.
+broken.  If *SUBSYSTEM* is supplied, this sets the logging level
+for any subsystem containing that string.  Subsystems include:
+
+* *lightningd*: The main lightning daemon
+* *database*: The database subsystem
+* *wallet*: The wallet subsystem
+* *gossipd*: The gossip daemon
+* *plugin-manager*: The plugin subsystem
+* *plugin-P*: Each plugin, P = plugin path without directory
+* *hsmd*: The secret-holding daemon
+* *gossipd*: The gossip daemon
+* *connectd*: The network connection daemon
+
+The following subsystems exist for each channel, where N is an incrementing
+internal integer id assigned for the lifetime of the channel:
+* *openingd-chan #N*: Each opening / idling daemon
+* *channeld-chan #N*: Each channel management daemon
+* *closingd-chan #N*: Each closing negotiation daemon
+* *onchaind-chan #N*: Each onchain close handling daemon
+* *json #FD*: Each JSONRPC connection, FD = file descriptor number
```

Comment:
From [line 188](https://github.com/ElementsProject/lightning/pull/3241/commits/9b6618e2c7534080e46b2e491ea353d6e59258f0#diff-6196cd48b2e3744d0cb85c6a997fbc2fR188) it seems that we just do a substring check. This is ok since otherwise we'd have to deal with `#` and spaces in a command line argument, however it also allows us to do some strange things like `--log-level=debug:an` and it'll set the level for `plugin-manager`, `openingd-chan`, `channeld-chan`, `closingd-chan`, and `onchaind-chan` which is a weird mix. Maybe we should just check for prefixes instead?

Reply:
You don't want only prefixes, because what if you want to watch channel #1 (or, as I was thinking of adding later, a specific peerid?).  You don't want to tie it to a particular daemon.

Reply:
It also explicitly says:

```markdown
If *SUBSYSTEM* is supplied, this sets the logging level for any subsystem containing that string.
```

I added the following paragraph:

```markdown
So, **log-level=debug:plugin** would set debug level logging on all
plugins and the plugin manager.  **log-level=io:chan #55** would set
IO logging on channel number 55 (or 550, for that matter).
```